tasks/coverage/snapshots/parser_misc.snap

```diff
@@ -1,7 +1,17 @@
 parser_misc Summary:
-AST Parsed     : 35/35 (100.00%)
-Positive Passed: 35/35 (100.00%)
+AST Parsed     : 36/36 (100.00%)
+Positive Passed: 35/36 (97.22%)
 Negative Passed: 33/33 (100.00%)
+Expect to Parse: tasks/coverage/misc/pass/oxc-10981.ts
+
+  × The keyword 'await' is reserved
+    ╭─[misc/pass/oxc-10981.ts:12:20]
+ 11 │     {
+ 12 │         for (using await of foo) {}
+    ·                    ─────
+ 13 │     }
+    ╰────
```

Comment:
interestingly, semantic incorreclty reports an error here:

https://www.typescriptlang.org/play/?target=99#code/N4KABBYK4M4JYDsDm17LAXjAewGYG4QBfEUSMXbAJzAApZEUH08dcBKMYEksySmvTQpK2Nm07diIXFAQBjAC5xsCMAEcoAUzqTwkPuQgC6zFAEMA7ubiLxoySXI8QILQA8ADtTvcgA

Reply:
I'll double check what's going on.

Reply:
I'll double check what's going on.

Reply:
tsc doesn't report any BindingIdentifier is `await` in module code errors.

babel: 

```
/repl.js: Unexpected reserved word 'await'. (5:19)

  3 | function que () {
  4 |     {  
> 5 |         for (using await of foo) {}
    |                    ^
  6 |     }
  7 | }
```